containers/src/Kokkos_ScatterView.hpp

```diff
@@ -532,32 +532,56 @@ void args_to_array(size_t* array, int pos, T dim0, Dims... dims) {
    subview where the index specified is the largest-stride one. */
 template <typename Layout, int rank, typename V, typename... Args>
 struct Slice {
-  using next       = Slice<Layout, rank - 1, V, Kokkos::ALL_t, Args...>;
-  using value_type = typename next::value_type;
-
-  static value_type get(V const& src, const size_t i, Args... args) {
+  using next = Slice<Layout, rank - 1, V, Kokkos::ALL_t, Args...>;
```

Comment:
This is just a remark, not a request for change.

In our code, we also have a `slice` utility but it seems not exactly for the same purpose. Ours is more like:
```c++
// take a subview, add ALL for remaining not-given indices up to rank
template <typename view_t, typename... Indices>
auto slice(view_t&& view, Indices&&... indices);

...

auto sliced = slive(my_view_of_rank_4, 1, 2); // equivalent to subview(my_view_of_rank_4, 1, 2, ALL, ALL);
```

It is a utility that could be thought of a bit more and promoted.

Reply:
This thing here is very ScatterView specific, because it sticks the all on the appropriate side. Beyond that this PR is solely targeted at overhauling ScatterView to allow for the View refactor.